tests/testthat/test-penalty.R

```diff
@@ -0,0 +1,19 @@
+context("penalty")
+
+test_that("we expect lasso solution when the non-convexity parameter is large enough", {
+  set.seed(2)
+
+  airquality <- na.omit(airquality)
+  x <- scale(as.matrix(subset(airquality, select = c("Wind", "Temp"))))
+  y <- scale(airquality$Ozone)
+
+  sgd_fit <- sgdnet(x, y, alpha = 1, thresh=1e-5)
+  sgd_mcp <- sgdnet(x, y, non_convexity = 1e+10, lambda = sgd_fit$lambda, thresh=1e-5, penalty = "MCP")
+  sgd_scad <- sgdnet(x, y, non_convexity = 1e+10, lambda = sgd_fit$lambda,thresh=1e-5,  penalty = "SCAD")
+
+  expect_equivalent(coef(sgd_fit), coef(sgd_mcp),
+                    tolerance = 0.001)
+
+  expect_equivalent(coef(sgd_fit), coef(sgd_scad),
+                    tolerance = 0.001)
```

Comment:
This looks good but can we use a tighter tolerance? We shouldn't struggle to get these comparable in a $p = 2$ case.